ql/src/java/org/apache/hadoop/hive/ql/exec/DDLTask.java

```diff
@@ -4522,6 +4522,12 @@ private void dropPartitions(Hive db, Table tbl, DropTableDesc dropTbl) throws Hi
       // for dropping. Thus, we need a way to push this filter (replicationSpec.allowEventReplacementInto)
       // to the  metastore to allow it to do drop a partition or not, depending on a Predicate on the
       // parameter key values.
+
+      if (tbl == null) {
+        // If table is missing, then partitions are also would've been dropped. Just no-op.
```

Comment:
i think this can happen only in case of same event getting replayed twice ??

Reply:
Nope. If same event applied twice, then second one will be no-op before reaching here as we validate the last repl id of database while building the DAG itself.
This case is possible if previous run of drop table event passed but setting last repl id failed due to metastore connection error or so. So, if you try to run this again, it may hit here. Anyways, I kept this check as a defensive code only.